framework/include/convergence/Convergence.h

```diff
@@ -37,10 +37,25 @@ class Convergence : public MooseObject,
     DIVERGED = -1
   };
 
+  /**
+   * Iteration type
+   */
+  enum class IterationType
+  {
+    NONLINEAR = 0,
+    LINEAR = 1, // solve_type = LINEAR, not linear solves within nonlinear solve
+    MULTISYSTEM_FIXED_POINT = 2,
+    MULTIAPP_FIXED_POINT = 3,
+    STEADY_STATE = 4
```

Comment:
we should consider either using a MooseEnum (and storing it somewhere) to which an app could add, OR adding a CUSTOM option for apps to leverage

Reply:
this could be an idea for later too btw

this isnt top priority rn